lib/systems/default.nix

```diff
@@ -49,12 +102,17 @@ rec {
       # Either of these can be losslessly-extracted from `parsed` iff parsing succeeds.
       system = parse.doubleFromSystem final.parsed;
       config = parse.tripleFromSystem final.parsed;
-      # Determine whether we can execute binaries built for the provided platform.
-      canExecute = platform:
-        final.isAndroid == platform.isAndroid &&
-        parse.isCompatible final.parsed.cpu platform.parsed.cpu
-        && final.parsed.kernel == platform.parsed.kernel;
-      isCompatible = _: throw "2022-05-23: isCompatible has been removed in favor of canExecute, refer to the 22.11 changelog for details";
+      isCompatible      = isCompatiblePlaceholder;
+
+      # 2023-06-17: `platform.canExecute` is deprecated in favor of `lib.systems.canExecute platform`
+      canExecute        = canExecutePlaceholder final;
+
+      # 2023-06-17: `platform.emulatorAvailable` has been removed in favor of `lib.systems.emulatorAvailable platform`
+      emulatorAvailable = emulatorAvailablePlaceholder final;
+
+      # 2023-06-17: `platform.emulator` has been removed in favor of `lib.systems.emulator platform`
+      emulator          = emulatorPlaceholder final;
```

Comment:
Won't this be overridden by the definition of `emulator` below?  (and likewise for `emulatorAvailable`)

Reply:
Thanks for noticing.  This PR is very annoying to rebase.

5dd051aa06a2